phpseclib/Net/SSH2.php

```diff
@@ -2132,9 +2144,11 @@ function _keyboard_interactive_process()
      */
     function _ssh_agent_login($username, $agent)
     {
+        $this->agent = $agent;
         $keys = $agent->requestIdentities();
         foreach ($keys as $key) {
             if ($this->_privatekey_login($username, $key)) {
+                $this->agent->_on_login_success($this);
```

Comment:
> calling a private/protected method from external scope

phpseclib already does that. See Net_SCP, for example. It calls `Net_SSH2::_get_channel_packet` for instance. I'm not really sure what the alternative is? public isn't a good level of visibility because it isn't intended to be public facing. I would not want it showing up in phpDocumentor generated code for instance. But at the same time neither private or protected are that great either since Net_SCP does access those methods.